src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -54,12 +52,7 @@ const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RB
       setIsFollow(true);
       try {
         await followTopic(topicName, isIncognito);
-        getTopicDetail(topicName);
-        const discoveryInitialTopicResponse = await DiscoveryRepo.fetchInitialDiscoveryTopics();
-        DiscoveryAction.setDiscoveryInitialTopics(
-          discoveryInitialTopicResponse.suggestedTopics as any,
-          discoveryDispatch
-        );
+        updateFollowTopicDiscoveryContext(true, {topic_id: topicId}, true);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new approach of updating the follow status in the discovery context using `updateFollowTopicDiscoveryContext` seems more efficient than fetching all initial discovery topics again. However, ensure that this function correctly updates the context and that there are no side effects due to not fetching all topics again.

```diff
-        const discoveryInitialTopicResponse = await DiscoveryRepo.fetchInitialDiscoveryTopics();
-        DiscoveryAction.setDiscoveryInitialTopics(
-          discoveryInitialTopicResponse.suggestedTopics as any,
-          discoveryDispatch
-        );
+        updateFollowTopicDiscoveryContext(true, {topic_id: topicId}, true);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->